controller/server/gui.ml

```diff
@@ -300,9 +300,39 @@ module NetworkGui = struct
     let%lwt () = Connman.Service.set_direct_proxy service in
     Lwt.return (success (Format.sprintf "Proxy of %s has been disabled." service.name))
 
+  (** Set static IP configuration on a service *)
+  let update_static_ip ~(connman: Connman.Manager.t) req =
+    let%lwt form_data = urlencoded_pairs_of_body req in
+    let%lwt service = with_service ~connman (param req "id") in
+    let get_prop s =
+      form_data
+      |> List.assoc s
+      |> List.hd
+    in
+    let address = get_prop "address" in
+    let netmask = get_prop "netmask" in
+    let gateway = get_prop "gateway" in
+    let nameservers = get_prop "nameservers" |> String.split_on_char ',' |> List.map (String.trim) in
+    let%lwt () = Connman.Service.set_ipv4 service  ~method':"manual" ~address ~netmask ~gateway in
+    let%lwt () = Connman.Service.set_nameservers service nameservers in
+    Lwt.return (success (Format.sprintf "Configured static IP for %s." service.name))
+
+  (** Remove static IP configuration from a service *)
+  let remove_static_ip ~(connman: Connman.Manager.t) req =
+    let%lwt form_data = urlencoded_pairs_of_body req in
+    let%lwt service = with_service ~connman (param req "id") in
+    let%lwt () = Connman.Service.set_ipv4 service  ~method':"dhcp" in
+    let%lwt () = Connman.Service.set_nameservers service [] in
+    Lwt.return (success (Format.sprintf "Removed static IP configuration of %s." service.name))
+
   (** Remove a service **)
   let remove ~(connman:Connman.Manager.t) req =
     let%lwt service = with_service ~connman (param req "id") in
+
+    (* Clear settings that might have been configured on the service. *)
+    let%lwt () = Connman.Service.set_nameservers service [] in
+    let%lwt () = Connman.Service.set_ipv4 service ~method':"dhcp" in
```

Comment:
I am wondering if we’d better move proxy removal here too, from the connect method:

```ocaml
(* Removing a proxy that would have been configured in the past *)
 let%lwt () = Connman.Service.set_direct_proxy service in
```

But in some cases, I am also wondering if it can happen that we don’t remove a connection manually, but it is unavailable, and we connect to another connection.

We could also move this clearing to the connection method.

What’s your thinking about this ?

Reply:
> But in some cases, I am also wondering if it can happen that we don’t remove a connection manually, but it is unavailable, and we connect to another connection.

I'm not completely certain what the significance of that would be. Should we have a quick call to discuss this?

Reply:
Done in https://github.com/dividat/playos/pull/66/commits/9bcdff5420e559e163b07a6a5211a86176ec82dd.
I didn't implement checking whether the resets are necessary, because a service doesn't have the `ipv4` property unless it's already connected.